File: containsDuplicate.java
import java.util.*;
class Solution{ // TC O(n) SC O(n)
	public boolean containsDuplicate(int[] nums){
		Set<Integer> set= new HashSet<>();
		for(int num: nums){
			if (set.contains(num))
				return true;
			set.add(num);
		}
		return false;
	}
	
}

class containsDuplicate{
	public static void main(String[] args) {
		int[] nums={1,2,3,1};
		Solution s= new Solution();
		System.out.println(s.containsDuplicate(nums));
	}
}

File: containsDuplicate.py
def containsDuplicate(nums): # TC O(n) // SC O(n)
	return len(set(nums)) < len(nums)

def containsDuplicate1(nums): # TC O(n) // SC O(n)
	seen=set()
	for num in nums:
		if num in seen:
			return True 
		seen.add(num)
	return False 

print(containsDuplicate([1,2,3,1]))
print(containsDuplicate1([1,2,3,1]))


